### ISSUE TITLE
Why is this awesome thing not working with webos 2.1

### ISSUE THREAD

**Issue by kurosch** [filmor/webos-messaging#11]
hey filmor,

this is more of a question than a problem.
I would really like to know why these plugins are not working with webos 2.1 and if possible how they would need reworking to be compatible.

Plus how would I need to proceed if I wanted to  port them to javascript.

**Comment by ericblade** [comment]
Porting the entire thing to Javascript sounds like crazy talk, in my opinion - you already have a native version, so I'm curious why one would want to do that?

While theoretically possible, it would be a lot easier to do if there are already node.js libraries that will handle some of that.  For example, I know that the best XMPP libs for Node also require native code, which supposedly we're not able to use with node services, however, I think that it does work, but I've never tested it.  There may just not be a ready system for compiling native node code for webOS.  

I don't personally have any idea on the question about 2.1, though. sorry. :)

**Comment by filmor** [comment]
Additional to what Eric said on part 2: You should not get the impression that I have more than a faint idea of how the actual protocols work. I took the code from palm and modified it to be more generic, but they also rely on libpurple (the library behind Pidgin) as the backend. What we could talk about would be to move some of the service code to a javascript service, since the current codebase is quite messy, but that would be a step to do for 2.0 while 1.0 isn't even finished ;)

On part 1: I actually don't know. It should work, I don't see how I could have used anything that is specific to webOS 2.2 or 3. Maybe I'll show up at some user group meeting to see if I could debug this on a live device :)
You can help however (as always) by providing logs of the installation process and running whatever is available.

**Comment by kurosch** [comment]
Part 2: So basically you are just hacking your way through libpurple for getting the messaging plugins done. Does that include that you can simply use those existing pidgin plugins?

**Comment by filmor** [comment]
The "hacking through" part has already been done by the Palm people. I made it more generic and rebuilt the "plumbing layer" (the stuff that the user actually sees), and yes, through this work it is able to use all existing pidgin plugins with little amount of work (basically you need to add a few files in the build script and one file to generate the account files), as long as they and their dependencies compile on ARM.

**Comment by kurosch** [comment]
alright did understand part 2 know. 

part1:
so I did install the plugins wanted to add an jabber account but I could not add any server only jabber id and the password. I do not beleave that is how it is supposed to be or am I wrong?

**Comment by filmor** [comment]
Well, it should show some options, this suggests that the service didn't start. Could you try to start it manually from the command line (i.e. wterm or novaterm)? You'd only need to enter `/var/usr/sbin/org.webosinternals.purple.validator` and post the output here (which will be a lot easier if you use novaterm ;)).

**Comment by kurosch** [comment]
I happen to get the following error message:

/var/usr/sbin/org.webosinternals.purple.validator: errore while loading shared libraries: libmojodb.so: cannot open shared object

**Comment by filmor** [comment]
That is actually very helpful :)
Do you by chance know if there is a similar library on webOS 2.1? Otherwise I could just provide a libmojodb built from the sources …

**Comment by kurosch** [comment]
I am not sure whether it exists but if I can just use the shell search command to find it tomorrow morning and I will use it to find it and report back. But otherwise would be nice to include libmojodb into it

**Comment by filmor** [comment]
You'd do me a favour if you could execute `ldd /var/usr/sbin/org.webosinternals.purple.validator` and `ldd /usr/bin/impurpleservice` and post the results here. I'm a bit confused, as the sources on http://opensource.palm.com are the same for webOS 2.0 and 2.1 (i.e. they both link against libmojodb).

**Comment by kurosch** [comment]
well I checked the the directories/files and the second one does not exist.
here is my console output:
https://www.dropbox.com/s/d6hkynlnqnkchvz/temp.png

**Comment by filmor** [comment]
Could you try it with imlibpurpletransport?

**Comment by kurosch** [comment]
result:
https://www.dropbox.com/s/z73lv7xoll49702/temp2.png

**Comment by filmor** [comment]
Okay … So it seems that on older devices mojodb was statically linked. I'll provide a second binary version that will be installed if libmojodb cannot be found.

**Comment by filmor** [comment]
libs.tar.bz2 contains a static libmojodb now. Not the cleanest solution, but as I have to provide libs anyway …

Reopen if this still doesn't work.

**Comment by kurosch** [comment]
Hey can you reopen the issue as it still does not work even after a reboot.
what data shall I send you?

**Comment by filmor** [comment]
Same as last time.

**Comment by kurosch** [comment]
if I can locate libmojodb on my device than you could just hard link it, right?

**Comment by filmor** [comment]
Getting the static library was not the problem (it is included in the palm source packages) and the binaries are indeed now linked against that static library (checked that), so if it's still not working there is another library missing. (Which would be very weird, as the previous investigation didn't show any other problem … :/)

Could you again provide the output of the commands given above?

**Comment by kurosch** [comment]
Well here is all the output from those required commands sorry for the bad formatting but I do not know how to get it right..

## /var/usr/sbin/org.webosinternals.purple.validator

/var/usr/sbin/org.webosinternals.purple.validator: error while loading shared libraries: libgnutls.so.26: cannot open shared object file: No such file or directory

## ldd /var/usr/sbin/org.webosinternals.purple.validator:

```
/usr/lib/libgoodabort.so (0x2aad3000)
/usr/lib/libmemcpy.so (0x2aadc000)
libpurple.so => /var/usr/lib/libpurple.so (0x2aae5000)
libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0x2abdd000)
libgobject-2.0.so.0 => /usr/lib/libgobject-2.0.so.0 (0x2acbd000)
libgmodule-2.0.so.0 => /usr/lib/libgmodule-2.0.so.0 (0x2ad01000)
liblunaservice.so => /usr/lib/liblunaservice.so (0x2ad0c000)
libmojoluna.so => /usr/lib/libmojoluna.so (0x2ad42000)
libmojocore.so => /usr/lib/libmojocore.so (0x2ad51000)
libcjson.so => /usr/lib/libcjson.so (0x2ad97000)
libxml2.so.2 => /usr/lib/libxml2.so.2 (0x2ada5000)
libgnutls.so.26 => not found
libresolv.so.2 => /lib/libresolv.so.2 (0x2aecd000)
libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x2aeea000)
libm.so.6 => /lib/libm.so.6 (0x2afc3000)
libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x2b03b000)
libc.so.6 => /lib/libc.so.6 (0x2b04e000)
libgnutls.so.26 => not found
librt.so.1 => /lib/librt.so.1 (0x2b18e000)
libdl.so.2 => /lib/libdl.so.2 (0x2b19e000)
libgthread-2.0.so.0 => /usr/lib/libgthread-2.0.so.0 (0x2b1ab000)
libmjson.so => /usr/lib/libmjson.so (0x2b1b7000)
libgoodfork.so.0 => /usr/lib/libgoodfork.so.0 (0x2b1c8000)
libpthread.so.0 => /lib/libpthread.so.0 (0x2b1d1000)
libPmLogLib.so => /usr/lib/libPmLogLib.so (0x2b1f1000)
/lib/ld-linux.so.3 (0x2aaab000)
```

ldd /usr/bin/impurpleservice does not exist however imlibpupletransport does and

##  ldd /ust/bin/imlibpurpletransport:

/usr/lib/libgoodabort.so (0x2aad3000)
    /usr/lib/libmemcpy.so (0x2aadc000)
    liblunaservice.so => /usr/lib/liblunaservice.so (0x2aaed000)
    libmojoluna.so => /usr/lib/libmojoluna.so (0x2ab23000)
    libmojocore.so => /usr/lib/libmojocore.so (0x2ab32000)
    libpurple.so.0 => /usr/lib/libpurple.so.0 (0x2ab78000)
    libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x2ac13000)
    libm.so.6 => /lib/libm.so.6 (0x2acec000)
    libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x2ad64000)
    libc.so.6 => /lib/libc.so.6 (0x2ad77000)
    libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0x2aeb7000)
    libpthread.so.0 => /lib/libpthread.so.0 (0x2af97000)
    libgthread-2.0.so.0 => /usr/lib/libgthread-2.0.so.0 (0x2afb7000)
    librt.so.1 => /lib/librt.so.1 (0x2afc3000)
    libcjson.so => /usr/lib/libcjson.so (0x2afd3000)
    libmjson.so => /usr/lib/libmjson.so (0x2afe1000)
    libgoodfork.so.0 => /usr/lib/libgoodfork.so.0 (0x2aff2000)
    libdl.so.2 => /lib/libdl.so.2 (0x2affb000)
    libPmLogLib.so => /usr/lib/libPmLogLib.so (0x2b008000)
    libgobject-2.0.so.0 => /usr/lib/libgobject-2.0.so.0 (0x2b015000)
    libgmodule-2.0.so.0 => /usr/lib/libgmodule-2.0.so.0 (0x2b059000)
    libxml2.so.2 => /usr/lib/libxml2.so.2 (0x2b064000)
    libutil.so.1 => /lib/libutil.so.1 (0x2b184000)
    libnsl.so.1 => /lib/libnsl.so.1 (0x2b191000)
    libresolv.so.2 => /lib/libresolv.so.2 (0x2b1b1000)
    /lib/ld-linux.so.3 (0x2aaab000)

This is the output without your plugin installed, though installed they are the same

**Comment by filmor** [comment]
Ah, didn't see that one coming :/

Okay, what I'll do until this weekend is, I will move from gnutls over to nss and provide this library as a so-file for that in the package. This should also fix #8 and might also work for #12.
